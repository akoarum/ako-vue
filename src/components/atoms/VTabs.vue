<template>
  <ul role="tablist" class="vTabs">
    <template v-for="tab in tabs">
      <li :key="tab.id" role="presentation" class="vTabs__item">
        <template v-if="link">
          <router-link :to="tab.to" role="tab" class="vTabs__tab">
            <span>{{ tab.label }}</span>
          </router-link>
        </template>
        <template v-else>
          <button
            type="button"
            role="tab"
            class="vTabs__tab"
            :aria-selected="String(tab.to === current)"
            :aria-controls="tab.to.replace(/^#(.*?)$/, '$1')"
            @click="$emit('click', tab.to)"
          >
            <span>{{ tab.label }}</span>
          </button>
        </template>
      </li>
    </template>
  </ul>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Tab } from '~/models/Tab'

export default Vue.extend({
  props: {
    tabs: { type: Array as PropType<Tab[]>, required: true },
    tag: { type: String, default: 'button' },
    current: { type: String, required: true },
    link: { type: Boolean }
  }
})
</script>

<style lang="scss" scoped>
.vTabs {
  display: flex;
}

.vTabs__tab {
  position: relative;
  font-size: 1.4rem;
  line-height: 1.4;
  transition: background 0.3s $easing;

  @include media-min {
    padding: 8px 16px 10px;

    &:hover,
    &:focus {
      background: #f0f0f0;
    }
  }

  @include media-max {
    padding: 8px 12px 10px;
  }
}

.vTabs__tab::before {
  position: absolute;
  bottom: 0;
  left: 0;
  content: '';
  width: 100%;
  height: 2px;
  background: $color-theme;
  opacity: 0;
  transition: opacity 0.3s $easing;
}

.vTabs__tab[aria-selected='true'] {
  color: $color-theme;
}

.vTabs__tab[aria-selected='true']::before {
  opacity: 1;
}
</style>
