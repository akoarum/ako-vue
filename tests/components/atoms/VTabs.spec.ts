import { shallowMount } from '@vue/test-utils'
import VTabs from '~/components/atoms/VTabs.vue'

describe('VTabs', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VTabs, {
      propsData: {
        tabs: [
          { id: 1, to: '#tab1', label: '項目1' },
          { id: 2, to: '#tab2', label: '項目2' },
          { id: 3, to: '#tab3', label: '項目3' }
        ],
        current: '#tab1'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
