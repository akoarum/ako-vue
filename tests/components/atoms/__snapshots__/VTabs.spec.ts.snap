// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTabs initialized snapshot 1`] = `
<ul
  class="vTabs"
  role="tablist"
>
  <li
    class="vTabs__item"
    role="presentation"
  >
    <button
      aria-controls="tab1"
      aria-selected="true"
      class="vTabs__tab"
      role="tab"
      type="button"
    >
      <span>
        項目1
      </span>
    </button>
  </li>
  <li
    class="vTabs__item"
    role="presentation"
  >
    <button
      aria-controls="tab2"
      aria-selected="false"
      class="vTabs__tab"
      role="tab"
      type="button"
    >
      <span>
        項目2
      </span>
    </button>
  </li>
  <li
    class="vTabs__item"
    role="presentation"
  >
    <button
      aria-controls="tab3"
      aria-selected="false"
      class="vTabs__tab"
      role="tab"
      type="button"
    >
      <span>
        項目3
      </span>
    </button>
  </li>
</ul>
`;
